### ISSUE TITLE
A lot of CSS files not found - Error 404

### ISSUE THREAD

**Issue by hldvuong** [OrchardCMS/Orchard#7932]
![error-404-not-found](https://user-images.githubusercontent.com/11913564/34663079-bd62e5c0-f44c-11e7-89df-2c7d2c796a66.png)

Please fix it !

**Comment by sebastienros** [comment]
Where is the page you are hitting? it should look for these css files in the relative modules.

**Comment by HengzheLi** [comment]
more information for troubleshooting.  

Those links only in Admin page. Seems that they are generated by code. And the more module is enabled the more link show up

**Comment by hldvuong** [comment]
I use Visual Studio 2017 clone the https://github.com/OrchardCMS/Orchard.git
And install
+ Use built-in data storage (SQL Server Compact) 
+ Choose Orchard Recipe : Blog

Login to admin -Dashboard:
http://localhost:8621/Admin
--> bug found here !

**Comment by sebastienros** [comment]
If you want to fix it, please create a PR that adds all the css file with an empty content.

**Comment by Hameds** [comment]
It seems that problem is in `dev` branch and `1.10.x` branch have those `*-admin.css`  files

**Comment by sebastienros** [comment]
Would be interesting to know at which point these files were removed. And that someone fixed it then, if there was no specific reason.

**Comment by HengzheLi** [comment]
I think this is not a file losing issue. It seems that it is the same as #3310. I found that those losted files are exist in the Styles folder of the Module that the file's name point to implicitly.

For example, the file "menu.users-admin.css" is under the Styles folder of Module Orchard.Users.
The file menu.settings-admin.css is under the folder of Orchard.Core/Settings/Styles

So it seems that paths of those files are caculated incorrectly.

**Comment by HengzheLi** [comment]
I think here is the reason. In Themes\TheAdmin\Views\Menu.cshtml  line 9.
```
    RegisterImageSet("menu." + imageSet, Model.MenuName /* style */, 16 /* bounding box */);
```
Then it goes to WebViewPage.cs to include a style file.
```
public void RegisterImageSet(string imageSet, string style = "", int size = 16) {
            // hack to fake the style "alternate" for now so we don't have to change stylesheet names when this is hooked up
            // todo: (heskew) deal in shapes so we have real alternates 
            var imageSetStylesheet = !string.IsNullOrWhiteSpace(style)
                ? string.Format("{0}-{1}.css", imageSet, style)
                : string.Format("{0}.css", imageSet);
            Style.Include(imageSetStylesheet);
        }
```
When the code goes here, the base folder of Style is Themes\TheAdmin\Styles, but the actual style files need to include are located in the style folder of Module where the style file belongs to.

I tied to fix this. But it is hard to know what is the module the style file belongs to.(Because my limited Orchard knowlage)

**Comment by sebastienros** [comment]
The actual issue is in the links being generated are not include /[Module] anymore.

Probably related to https://github.com/OrchardCMS/Orchard/issues/8008 so fixing it should fix both issues.

**Comment by HengzheLi** [comment]
unfortunately, to fix #8008 could not fix this issue. I already apply #8009 which has fixed #8008 in my local version. But this issue still exist.

**Comment by niranjankala** [comment]
I am in the progress to upgrade the latest version for client website but i am also facing this issue in Dev branch so which branch is stable to checkout?

**Comment by sebastienros** [comment]
@niranjankala 1.10.x is the most stable. dev has new features and breaking changes.

**Comment by barthamark** [comment]
This feature broke with this PR merged #7564 into dev which fixed #3954 .

The problem is that in StylesheetBindingStrategy.cs Discover() the virtual path is ignored if the shape.Url has value (which might contain the query string). Using this value fixes the query string problem but ignores the hit.fileVirtualPath which points to the correct filename (the one in the module's folder instead of the folder in TheAdmin theme).

It seems that it fixes #7754 as well.

**Comment by BenedekFarkas** [comment]
Fixed with #8199.
